Composer/packages/client/src/router.tsx

```diff
@@ -141,14 +141,22 @@ const ProjectRouter: React.FC<RouteComponentProps<{ projectId: string; skillId:
         setLocale(storedLocale, rootBotProjectId);
       }
     }
-  }, [botProjectSpaceLoaded, rootBotProjectId]);
+  }, [botProjectSpaceLoaded, rootBotProjectId, botProjects]);
 
   useEffect(() => {
     if (props.projectId && !botProjects.includes(props.projectId)) {
       fetchProjectById(props.projectId);
     }
   }, [props.projectId]);
 
+  useEffect(() => {
+    if (props.skillId) {
+      setCurrentProjectId(props.skillId);
+    } else {
+      setCurrentProjectId(projectId);
```

Comment:
either this should be props.projectId or the dependency should be projectId instead of props.projectId

Reply:
updated to props.projectId